lib/app/endpoint/terminology.rb

```diff
@@ -86,43 +93,45 @@ def find_param(parameters, name)
         parameters.parameter.detect { |param| param.name == name }&.value
       end
 
-      def normalize_valueset_coding(parameters)
-        if find_param(parameters, 'coding')
-          find_param(parameters, 'coding').to_hash
-        elsif find_param(parameters, 'system') && find_param(parameters, 'code')
-          { 'code' => find_param(parameters, 'code'), 'system' => find_param(parameters, 'system') }
-        else
-          {}
-        end
-      end
-
-      def normalize_codesystem_coding(parameters, system_param = nil)
+      def normalize_coding(parameters, system_param = nil)
         if find_param(parameters, 'coding')
-          find_param(parameters, 'coding').to_hash
-        elsif (find_param(parameters, 'url') || system_param) && find_param(parameters, 'code')
-          system_param ||= find_param(parameters, 'url')
-          { 'code' => find_param(parameters, 'code'), 'system' => system_param }
+          [find_param(parameters, 'coding').to_hash]
+        elsif system_param && find_param(parameters, 'code')
+          [{ 'code' => find_param(parameters, 'code'), 'system' => system_param }]
+        elsif find_param(parameters, 'codeableConcept')
+          # NOTE: This branch doesn't wrap the response in an array because the 'coding' field on a CodeableConcept is already an array
+          find_param(parameters, 'codeableConcept').to_hash['coding']
         else
-          {}
+          []
         end
       end
 
       def codesystem_validates_code(parameters, id_param)
-        coding = normalize_codesystem_coding(parameters, id_param)
-        validation_fn = FHIR::StructureDefinition.vs_validators[coding['system']]
+        id_param ||= find_param(parameters, 'url')
+        coding_arr = normalize_coding(parameters, id_param)
+        validation_fn = FHIR::StructureDefinition.vs_validators[coding_arr.first['system']]
         if validation_fn
           # NOTE: This function does not yet validate `display` attributes
-          retval = if validation_fn.call(coding)
-                     FHIR::Parameters.new(parameter: [FHIR::Parameters::Parameter.new(name: 'result', valueBoolean: true)])
-                   else
-                     message = "The code '#{coding['code']}' is not a valid member of '#{coding['system']}'"
-                     params = [
-                       FHIR::Parameters::Parameter.new(name: 'result', valueBoolean: false),
-                       FHIR::Parameters::Parameter.new(name: 'message', valueString: message)
-                     ]
-                     FHIR::Parameters.new(parameter: params)
-                   end
-          respond_with_type(retval, request.accept, 200)
+          codes_valid_arr = coding_arr.map do |coding|
```

Comment:
everything contained in `if validation_fn ...` seems to be the same as for `valueset_validates_code`.  Could be broken out into a shared function unless I just missed something that makes them different.